pkg/istio-agent/xds_proxy.go

```diff
@@ -464,7 +473,11 @@ func (p *XdsProxy) handleUpstreamResponse(con *ProxyConnection) {
 					forwardToEnvoy(con, resp)
 				}
 			default:
-				forwardToEnvoy(con, resp)
+				if strings.HasPrefix(resp.TypeUrl, "istio.io/debug/") {
```

Comment:
Would it make sense to use DiscoveryRequest.ResponseNonce, and have the DiscoveryResponse.Nonce contain a matching value?  I realize this is backwards from how it is used, but it would let the requests and replies be matched precisely.

Reply:
It would be great if this change involved no changes to pilot's code at all. If that is impossible we can compromise, but ideally its all client side. Maybe resourceName could encode it? I will think on this

Reply:
With the latest, the only change to Pilot is to use Event Generator instead of API Generator.